### ISSUE TITLE
💅 noProcessEnv should report env imported from node:process

### ISSUE THREAD

**Issue by dBianchii** [biomejs/biome#10447]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.15
  Color support:                false

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                    unset
  BIOME_WATCHER_KIND:               unset
  BIOME_WATCHER_POLLING_INTERVAL:   unset
  NO_COLOR:                     1
  TERM:                         dumb
  JS_RUNTIME_VERSION:           v24.12.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.13

Biome Configuration:
  Status:                       Loaded successfully
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               false

Linter:
  Enabled rules:
    style/noProcessEnv
```

</details>

### Rule name

`style/noProcessEnv`

### Playground link

I could not produce a useful playground link for this because the reproduction depends on `node:process` imports and a minimal `noProcessEnv` config. Minimal reproduction is below.

### Reproduction

Use this config:

```json
{
	"formatter": { "enabled": false },
	"assist": { "enabled": false },
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": false,
			"style": {
				"noProcessEnv": "error"
			}
		}
	}
}
```

Given these files:

```ts
// direct.ts
console.log(process.env.FOO);
```

```ts
// import-default.ts
import process from "node:process";

console.log(process.env.FOO);
```

```ts
// import-env.ts
import { env } from "node:process";

console.log(env.FOO);
```

```ts
// bun-env.ts
console.log(Bun.env.FOO);
```

Run:

```bash
biome check direct.ts import-default.ts import-env.ts bun-env.ts
```

Current result:

- `direct.ts` reports `lint/style/noProcessEnv`.
- `import-default.ts` reports `lint/style/noProcessEnv`.
- `import-env.ts` does not report anything.
- `bun-env.ts` does not report anything.

### Expected result

`style/noProcessEnv` should also report direct environment reads through common runtime aliases, including `env` imported from `node:process` and `Bun.env`.

For example:

```ts
import { env } from "node:process";

console.log(env.FOO);
```

```ts
console.log(Bun.env.FOO);
```

From a rule-consumer perspective these are still environment reads. Today, projects that want to consistently ban direct environment reads need to add separate `noRestrictedImports` and `noRestrictedGlobals` entries to prevent `import { env } from "node:process"` and `Bun.env`, even though `noProcessEnv` is the rule that semantically owns this restriction.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by xxiaoxiong** [comment]
I've implemented this in PR #10452.

The rule now detects:
- ✅ `env` imported from `'node:process'` or `'process'`
- ✅ `Bun.env` usage

Test cases added for all patterns including valid cases that should not trigger.

**Comment by dBianchii** [comment]
This still reproduces for me on `@biomejs/biome@2.5.0`.

With only `style/noProcessEnv` enabled, this still passes without a diagnostic:

```ts
import { env } from "node:process";

const fromNodeProcessImport = env["TEST_ENV"];

void fromNodeProcessImport;
```

`Bun.env` also still passes without a diagnostic:

```ts
const fromBun = Bun.env["NODE_ENV"];

void fromBun;
```

Current workaround in consumer config is to add `noRestrictedImports` entries for `node:process`, `process`, and `bun` named `env` imports, but there does not appear to be a Biome config-only workaround for direct `Bun.env[...]` without banning the whole `Bun` global.

**Comment by ematipico** [comment]
I argue that this should be a different rule. The rule is meant to catch `process.env`, and `Bun.env` is different

**Comment by dBianchii** [comment]
That split makes sense. I separated the follow-ups:

- Node/process alias case: #10619 (`import { env } from "node:process"` / `"process"` should still be handled by `noProcessEnv`)
- Bun-specific case: #10617 (separate request for a rule/config support to report direct `Bun.env` reads)

So this closed issue can stay scoped to the original process env alias behavior, and the Bun discussion can continue separately.
